src/packed_seq.rs

```diff
@@ -506,17 +506,31 @@ impl SeqVec for PackedSeqVec {
     /// TODO: Optimize for non-BMI2 platforms.
     /// TODO: Support multiple ways of dealing with non-`ACTG` characters.
     fn push_ascii(&mut self, seq: &[u8]) -> Range<usize> {
-        let start = 4 * self.seq.len();
+        let start_aligned = 4 * self.seq.len();
+        let start = self.len;
         let len = seq.len();
 
+        let unaligned = core::cmp::min(start_aligned - start, len);
+        if unaligned > 0 {
+            // self.seq cannot be empty when start_aligned > start
+            let mut packed_byte = unsafe { *self.seq.last().unwrap_unchecked() };
+            for &base in &seq[..unaligned] {
+                packed_byte |= pack_char(base) << ((self.len % 4) * 2);
+                self.len += 1;
+            }
+            unsafe {
```

Comment:
Add a test where you push many pieces of random length 0 to 8, and check that in the end everything works out.
I think you're good here because this write here *always* triggers, even when this byte isn't completely filled, but still. (Also test correctness of the returned ranges then.)